handler/handler.go

```diff
@@ -0,0 +1,57 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ONSdigital/dp-search-data-importer/event"
+	"github.com/ONSdigital/dp-search-data-importer/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+//go:generate moq -out eventtest/result_writer.go -pkg eventtest . ResultWriter
+
+var _ event.Handler = (*BatchHandler)(nil)
+
+// BatchHandler handles batches of SearchDataImportModel events that contain CSV row data.
+type BatchHandler struct {
+	resultWriter ResultWriter
+}
+
+// TODO : ResultWriter dependency that outputs results
+type ResultWriter interface{}
+
+// NewBatchHandler returns a resultWriter TBD.
+func NewBatchHandler(
```

Comment:
I don't think this is needed anymore either? - we are not writing to temp file on disk anymore

Reply:
this is needed for running unit tests